source/common/router/router.cc

```diff
@@ -444,6 +444,14 @@ void Filter::chargeUpstreamCode(Http::Code code,
   chargeUpstreamCode(response_status_code, *fake_response_headers, upstream_host, dropped);
 }
 
+absl::optional<bool> getSampleValueFromShadowPolicy(const ShadowPolicy& shadow_policy) {
+  // If trace sampling is not explicitly configured in shadow_policy, we pass null optional to
+  // inherit the parent's sampling decision. This prevents oversampling when runtime sampling is
+  // disabled.
+  return shadow_policy.hasTraceSampledConf() ? absl::make_optional(shadow_policy.traceSampled())
+                                             : absl::nullopt;
+}
```

Comment:
I am good to the new default behavior. But please add a runtime flag to guard it.

Reply:
I added a runtime flag. Thanks for the suggestion.